app/models/order.rb

```diff
@@ -1,4 +1,10 @@
 class Order < ApplicationRecord
+  serialize :reward_snapshot
+
   belongs_to :employee
   belongs_to :reward
+
+  def snapshot_price
```

Comment:
other classes that will be calling that method should not really know or care, that it is a snapshot, so I would rename it to something a bit more descriptive :)

```suggestion
  def purchase_price
```